file-attribute-caching/src/main/kotlin/com/pkware/filesystem/attributecaching/AttributeCachingPath.kt

```diff
@@ -37,88 +39,116 @@ public const val CACHE_KEY_POSIX: String = "posix:*"
 public const val CACHE_KEY_ACL: String = "acl:*"
 
 /**
- * A [Path] instance that supports caching of [BasicFileAttributes] and other classes that extend it such as
- * [DosFileAttributes] and [PosixFileAttributes].
+ * A [Path] instance that supports caching of [BasicFileAttributes] along with other classes such as
+ * [DosFileAttributes], [PosixFileAttributes], ACL (access control list) [UserPrincipal] owner, and ACL
+ * (access control list) [AclEntry].
  *
  * @param fileSystem the [FileSystem] associated with this [AttributeCachingPath] instance.
  * @param delegate the [Path] to forward calls to if needed.
- * @param initializeCache `true` to initialize the cached attribute fields of this [AttributeCachingPath] instance.
- * The default is `false`.
  */
 internal class AttributeCachingPath(
     private val fileSystem: FileSystem,
     internal val delegate: Path,
-    private val initializeCache: Boolean = false,
 ) : ForwardingPath(delegate) {
 
+    private val delegateSupportedFileAttributeViews = delegate.fileSystem.supportedFileAttributeViews()
+
     /**
-     * Status variable to tell outside callers if this path has been initialized or not.
+     * Shows whether [BasicFileAttributes] are cached. `true` for cached, default is
+     * `false`.
      */
-    private var isInitialized = false
-
-    private val delegateSupportedFileAttributeViews = delegate.fileSystem.supportedFileAttributeViews()
+    var basicAttributesCached = false
+        private set
 
     private var basicAttributes by LazyAttribute {
-        delegate.fileSystem.provider().readAttributes(delegate, BasicFileAttributes::class.java)
+        val basicFileAttributes = delegate.fileSystem.provider().readAttributes(
+            delegate,
+            BasicFileAttributes::class.java,
+        )
+        basicAttributesCached = true
```

Comment:
Rather than track named variables to all of these, it would be much cleaner to add a property to the `LazyAttribute` that handles it. That way, you allow the class to handle the conditional logic rather than callers of the class.

Reply:
question with this is how to i keep the attribute itself private and the cached boolean property public with a private set?

Reply:
> Yeah I agree. I don't think you would be able to do the by syntax anymore but just converting it ultimately to a wrapper would pretty much do everything we want.

I think Ryan is correct about this.

> question with this is how to i keep the lazy attribute itself private and the cached boolean property public with a private set?

If it's a class, I don't think you need to expose the variable. I don't even think you'd need the variable.

```
val someAttr = LazyAttriibute({ yourInit() })

...

target.someAttr = someAttr // doesn't trigger autoload, internally knows whether or not it's cached
```

Reply:
> If it's a class, I don't think you need to expose the variable. I don't even think you'd need the variable.

To be clear, I'm saying if you make this a wrapper, not keep it so that it's still of type `BasicFileAttributes` or whatever.